scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -331,6 +445,19 @@
 
     api-client))
 
+(defn guard-invalid-synthetic-pods-config
+  "If the synthetic pods configuration section is present, validates it (throws if invalid)"
+  [compute-cluster-name synthetic-pods-config]
+  (when synthetic-pods-config
+    (when-not (and
+                (-> synthetic-pods-config :image count pos?)
+                (-> synthetic-pods-config :user count pos?)
+                (-> synthetic-pods-config :max-pods-outstanding pos?)
+                (-> synthetic-pods-config :pools set?)
+                (-> synthetic-pods-config :pools count pos?))
+      (throw (ex-info (str "In " compute-cluster-name " compute cluster, invalid synthetic pods config")
+                      synthetic-pods-config)))))
```

Comment:
Can we use `schema` for the validation or pre-conditions? It will make it easier to detect which validation failed.

Reply:
Will do